src/providers/twitch/api/Helix.cpp

```diff
@@ -544,40 +548,30 @@ void Helix::createStreamMarker(
 };
 
 void Helix::loadBlocks(QString userId,
-                       ResultCallback<std::vector<HelixBlock>> successCallback,
-                       HelixFailureCallback failureCallback)
+                       ResultCallback<std::vector<HelixBlock>> pageCallback,
+                       FailureCallback<QString> failureCallback,
+                       CancellationToken &&token)
 {
-    QUrlQuery urlQuery;
-    urlQuery.addQueryItem("broadcaster_id", userId);
-    urlQuery.addQueryItem("first", "100");
-
-    this->makeGet("users/blocks", urlQuery)
-        .onSuccess([successCallback, failureCallback](auto result) -> Outcome {
-            auto root = result.parseJson();
-            auto data = root.value("data");
-
-            if (!data.isArray())
-            {
-                failureCallback();
-                return Failure;
-            }
+    this->paginate(
+        u"users/blocks"_s,
+        {{u"broadcaster_id"_s, userId}, {u"first"_s, u"100"_s}},
+        [pageCallback](const QJsonObject &json) {
+            const auto data = json["data"_L1].toArray();
```

Comment:
If the array is empty (i.e. it wasn't a valid array, or it was actually an empty array), make this function return something to tell paginate to stop paginating